src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -794,9 +814,14 @@ protected ArrayBuilder<Symbol> AddOrWrapTupleMembers(ImmutableArray<Symbol> curr
             }
 
             elementsMatchedByFields.Free();
+            members.AddRange(nonFieldMembers);
+            nonFieldMembers.Free();
+            if (fieldDefinitionsToIndexMap is object)
+            {
+                this.TupleData!.SetFieldDefinitionsToIndexMap(fieldDefinitionsToIndexMap);
```

Comment:
>fieldDefinitionsToIndexMap [](start = 62, length = 26)

Could we end up in a situation when member list comes from one thread and the map from another, effectively based on different sets of symbols?